Team_47/QuizCreator/src/main/java/com/asu/ser516/team47/servlet/QuizCreationServlet.java

```diff
@@ -0,0 +1,256 @@
+
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.util.*;
+
+/**
+ * Endpoint for professor to create a quiz.
+ *
+ * @author David Lahtinen
+ * @version 1.0
+ * @since 02/28/19
+ */
+@WebServlet(name = "QuizCreationServlet")
+public class QuizCreationServlet extends HttpServlet {
+    //Garbage value is used where needed for constructors where the value doesn't matter.
+    final private int GARBAGE_VALUE = -1;
+    private String title = null;
+    private Integer course_id = null;
+    private String instructions = null;
+    private Boolean shuffle = null;
+    private Integer time_limit = null;
+    private Date date_open = null;
+    private Date date_close = null;
+    private String quiz_type = null;
+    private Integer attempts = null;
+    private String quiz_group = null;
+    private Double total_points = null;
+    private JSONArray jsonQuestions = null;
+    private List<Question> questionList = new ArrayList<>();
+    private List<List<Choice>> choiceTable = new ArrayList<>();
+
+    /**
+     * Take newly submitted quiz from professor and insert into database.
+     *
+     * @param req HttpRequest containing json form of quiz data
+     * @param res HttpResponse
+     * @throws IOException
+     */
+    @Override
+    public void doPost(HttpServletRequest req, HttpServletResponse res) throws IOException {
+        JSONObject jsonQuiz = null;
+
+        //Validate all required fields are present
+        try {
+            jsonQuiz = JSONRequestParser.getJsonFromRequest(req);
+        } catch (ParseException pe) {
+            res.sendError(400, "Could not parse form");
+            return;
+        }
+        if (!validateQuizFields(jsonQuiz)){
+            res.sendError(400, "Invalid fields.");
+            return;
+        }
+        //as close date is optional, verify separately
+        try {
+            String dateStr = (String) jsonQuiz.get("date_open");
+            Calendar cal = javax.xml.bind.DatatypeConverter.parseDateTime(dateStr);
+            date_open = cal.getTime();
+            dateStr = (String) jsonQuiz.get("date_close");
+            cal = javax.xml.bind.DatatypeConverter.parseDateTime(dateStr);
+            date_close = cal.getTime();
+            if (date_open.after(date_close)) {
+                res.sendError(400, "Nice try, you monster.");
+                return;
+            }
+        } catch (ClassCastException ex) {
+            res.sendError(400, "Invalid format for dates.");
+            return;
+        } catch (NullPointerException npe) {
+            //Date not included: open immediately and use no close date.
+        }
+        if (date_open == null) date_open = new Date();
+        Quiz qz = new Quiz(GARBAGE_VALUE, title, course_id, instructions, shuffle, time_limit, date_open, date_close,
```

Comment:
title and course_id need to be switched. As it stands this does not compile.

Reply:
That is embarrasing; I thought I compiled this before making a PR.